### PR TITLE
Add test case for federated jit provisioning with association

### PR DESCRIPTION
**Pull request by darshanasbg**
Test case for https://github.com/wso2/product-is/issues/4207

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/provisioning/JustInTimeProvisioningTestCase.java [hunk 1]
```diff
@@ -168,6 +169,18 @@ public void testSAMLToSAMLFederationWithAsIsUserNameUserStore() throws Exception
...
     }
 
+    @Test(priority = 8, groups = "wso2.is", description = "test Just in time provisioning with association")
+    public void testSAMLToSAMLFederationWithAssociation() throws Exception {
+
+        updateIdentityProviderJitConfiguration(false, false, false, UserCoreConstants.PRIMARY_DEFAULT_DOMAIN_NAME);
+        updateServiceProviderAssociationConfiguration(true);
+        super.testSAMLToSAMLFederation(false);
+
+
```
Comment (JKAUSHALYA) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/provisioning/JustInTimeProvisioningTestCase.java]: Why two new lines ?

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/saml/SAMLIdentityFederationTestCase.java [hunk 2]
```diff
@@ -286,6 +286,12 @@ public void testAttributeConsumerIndex() throws Exception {
     }
 
     public void testSAMLToSAMLFederation() throws Exception {
+        testSAMLToSAMLFederation(true);
```
Comment (JKAUSHALYA) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/saml/SAMLIdentityFederationTestCase.java]: Needs a new line after method signature.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/saml/SAMLIdentityFederationTestCase.java [hunk 3]
```diff
@@ -643,7 +652,7 @@ public boolean validateSAMLResponse(HttpResponse response, String userName) thro
         }
         bufferedReader.close();
         assertLocalClaims(buffer.toString());
-        return buffer.toString().contains("You are logged in as " + userName);
+        return buffer.toString().contains("<h2> You are logged in as " + userName + "</h2>");
     }
 
 }
```
Comment (JKAUSHALYA) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/saml/SAMLIdentityFederationTestCase.java]: New line at the end.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/provisioning/JustInTimeProvisioningTestCase.java [hunk 4]
```diff
@@ -168,6 +169,18 @@ public void testSAMLToSAMLFederationWithAsIsUserNameUserStore() throws Exception
...
+        Assert.assertTrue(userStoreClient.isExistingUser(getFederatedTestUser()));
+        userStoreClient.deleteUser(getFederatedTestUser());
```
Comment (JKAUSHALYA) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/provisioning/JustInTimeProvisioningTestCase.java]: Shall we move this to a "finally" block or "AfterClass" anotated method?

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/provisioning/JustInTimeProvisioningTestCase.java [hunk 5]
```diff
@@ -204,6 +217,20 @@ private void updateIdentityProviderJitConfiguration(boolean isPasswordProvisioni
         super.updateIdentityProvider(PORT_OFFSET_0, IDENTITY_PROVIDER_NAME, identityProvider);
     }
 
+    /**
+     * To update the Service provider association configuration.
+     *
+     * @param isModifyUserNameAllowed To mention whether isModifyUserNameAllowed.
+     * @throws Exception Exception.
+     */
+    private void updateServiceProviderAssociationConfiguration(boolean assertIdentityUsingMappedLocalSubjectIdentifierEnabled) throws Exception {
```
Comment (vihanga-liyanage) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/provisioning/JustInTimeProvisioningTestCase.java]: Shall we limit 120 characters per line?

**Pull request merged**